hr_worked_days_from_timesheet/models/hr_payslip.py

```diff
@@ -0,0 +1,97 @@
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def _timesheet_mapping(self, timesheet_sheets, payslip, date_from,
+                           date_to):
+        """This function takes timesheet objects imported from the timesheet
+        module and creates a dict of worked days to be created in the payslip.
+        """
+        # Create one worked days record for each timesheet sheet
+        wd_model = self.env['hr.payslip.worked_days']
+        for ts_sheet in timesheet_sheets:
+            # Get formated date from the timesheet sheet
+            date_from_formated = fields.Date.to_string(
+                fields.Datetime.from_string(ts_sheet.date_from))
+            number_of_hours = 0
+            for ts in ts_sheet.timesheet_ids:
+                if date_from <= ts.date <= date_to:
+                    number_of_hours += ts.unit_amount
+
+            if number_of_hours > 0:
+                wd_model.create({
+                    'name': _('Timesheet %s') % date_from_formated,
+                    'number_of_hours': number_of_hours,
+                    'contract_id': payslip.contract_id.id,
+                    'code': 'TS',
+                    'imported_from_timesheet': True,
+                    'timesheet_sheet_id': ts_sheet.id,
+                    'payslip_id': payslip.id,
+                })
+
+    @api.multi
+    def import_worked_days(self):
+        """This method retreives the employee's timesheets for a payslip period
+        and creates worked days records from the imported timesheets
+        """
+        for payslip in self:
+
+            date_from = payslip.date_from
+            date_to = payslip.date_to
+
+            # Delete old imported worked_days
+            # The reason to delete these records is that the user may make
+            # corrections to his timesheets and then reimport these.
+            wd_model = self.env['hr.payslip.worked_days']
+            wd_model.search(
+                [('payslip_id', '=', payslip.id),
+                 ('imported_from_timesheet', '=', True)]).unlink()
+
+            # get timesheet sheets of employee
+            timesheet_sheets = self.env['hr_timesheet_sheet.sheet'].search(
+                ['&', '&', '|', '&',
+                 ('date_from', '<=', date_to), ('date_from', '>=', date_from),
+                 '&', ('date_to', '<=', date_to), ('date_to', '>=', date_from),
+                 ('state', '=', 'done'),
+                 ('employee_id', '=', payslip.employee_id.id)]
+            )
+
+            if not timesheet_sheets:
+                raise Exception(
```

Comment:
Could you use exception.Warning from openerp.exception, if we leave just python Exception it's going to throw an Odoo Server Error instead a Warnign to the user...